drivers/volume/portworx/portworx.go

```diff
@@ -1026,7 +1051,7 @@ func (p *portworx) StartVolumeSnapshotRestore(snapRestore *stork_crd.VolumeSnaps
 				}
 			}
 			replNodes := vols[0].GetReplicaSets()[0]
-			taskID := restoreTaskPrefix + snapID + "-" + volID
+			taskID := restoreTaskPrefix + snapID + "-" + volID + string(snapRestore.GetUID())
```

Comment:
Generate the taskID and volume name in a separate function and use them here and in `CleanupSnapshotRestoreObjects`

Reply:
Don't need the `snapID` in `taskID`, it is pretty long. `volID + restoreUID` should make it unique